hw_11/HW_1_Sotnikov.py

```diff
@@ -0,0 +1,368 @@
+        last_message = self.chat_history[number]
+        print(f"User info: {last_message.show('user')}\n\
+        Date: {last_message.show('date')}\n\
+        Time: {last_message.show('time')}\n\
+        Text of the message: {last_message.show('text')}")
+
+
+    def get_history_from_time_period(self, first_date=None, second_date=None):   # Функция работает криво
+        start_point = None
+        final_point = None
+        Flag_start = True
+
+        if first_date != None or second_date != None:
+            for message in self.chat_history:
+
+                if (second_date - message.datetime).days >= 0 \
+                and (message.datetime - first_date).days >= 0 \
+                and Flag_start:
+                    start_point = self.chat_history.index(message)
+                    Flag_start = False
+
+                if (message.datetime - first_date).days < 0 and Flag_start == False:
+                    final_point = self.chat_history.index(message)
+                    break
+
+        if first_date == None:
+            start_point = 0
+
+        if second_date == None:
+            final_point = len(self.chat_history)
+
+
+        result = self[start_point:final_point]
+        return result
+
+
+    def show_chat(self):
+        for message_number in range(len(self.chat_history)):
+            self.show_last_message(message_number)
+
+    def recieve(self, message):
+         if isinstance(message, Message):
+            self.chat_history.insert(0, message)
+            self.insert(0, message)
+
+## Creating class Message
+class Message:
+    def __init__(self, text, user, datetime=None):
+        self.text = text
+        self.datetime = datetime
+        self.date = None
+        self.time = None
+        self.user = user
+
+    def show(self, attribute):
+        if isinstance(attribute, list):
+            answers = []
+            for attr in attribute:
+                if isinstance(attr, str) and hasattr(self, attr):
+                    answers.append(f"{attr}: {getattr(self, attr)}")
+                else:
+                    raise AttributeError\
+                    ("You have entered unavailable argument.\n\
+                    Please input one of the below-writed attributes or list/tuple with them:\n\
+                    'user', 'date', 'time', 'message'")
+            return answers
+        elif isinstance(attribute, str) or isinstance(attribute, User):
+            if hasattr(self, attribute):
+                return f"{attribute}: {getattr(self, attribute)}"
+            else:
+                raise AttributeError\
+            ("You have entered unavailable argument.\n\
+            Please, input one of the below-writed attributes or list/tuple with them:\n\
+            'user', 'date', 'time', 'message'")
+        else:
+            raise AttributeError\
+            ("You have used unavailabel class of argument\n\
+            Please, use string type for message and User type for user.")
```

Comment:
Круто, что описаны разные варианты, но ожидалась здесь какая-то простая штука а-ля

```suggestion
        print(f"{self.time} {self.user.name}: {self.text}")
```